app/src/main/java/com/example/fisrtapplication/utils/ApplicationEx.java

```diff
@@ -0,0 +1,52 @@
+package com.example.fisrtapplication.utils;
+
+import android.app.Application;
+import android.content.Intent;
+
+import com.example.fisrtapplication.activities.DataListActivity;
+import com.example.fisrtapplication.api.VendingApiClient;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class ApplicationEx extends Application {
+
+    private FirebaseAuth mAuth;
+    private VendingApiClient vendingApiClient;
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+
+        mAuth = FirebaseAuth.getInstance();
+        vendingApiClient = createVendingApiClient();
+
+        staySigned();
+    }
+
+    public FirebaseAuth getAuth() {
+        return mAuth;
+    }
+
+    public void staySigned() {
```

Comment:
Please remove this method. Its potentially buggy and will create lots of issue after you'll migrate to fragment navigation